### DISCUSSION TITLE
Example recipe/config for pruned RNN-T loss ?

### DISCUSSION THREAD

**Discussion by amisra001** [espnet/espnet#5681] [General]
Hi, is there a config file or recipe using the  ESPnetASRTransducerModel? Most of the transducer results are based on ESPnetASRModel class. **I'm not able to replicate my RNN-T results that i got using ESPnetASRModel class when i try to use asr_transducer.py script**. I want to use asr_transducer task mainly because of k2 based RNN-T loss.  @b-flo

**Comment by b-flo** [comment]
Hi,

Sorry for the delay!

> Hi, is there a config file or recipe using the ESPnetASRTransducerModel? 

There are several. Not sure which one fit your dataset but you can take the [Librispech](https://github.com/espnet/espnet/tree/master/egs2/librispeech/asr1/conf/tuning/transducer) config as an example.
For pruned RNN-T loss, you only need to set `use_k2_pruned_loss: True` in `model_conf`. For further options, refer to the documentation.

**Comment by amisra001** [comment]
Hi @b-flo  thanks for your reply. I think the config file you mentioned uses ESPnetASRModel and not  ESPnetASRTransducerModel. In fact, most of the recipes are configured to use ESPnetASRModel class. To use  ESPnetASRTransducerModel class i followed instructions mentioned here: https://github.com/espnet/espnet/blob/master/doc/espnet2_tutorial.md 

I created a config file that ESPnetASRTransducerModel class can accept but didn't get same results as using the ESPnetASRModel class on the same dataset. 

Right now, I'm modifying ESPnetASRModel class to accept k2 pruned loss parameter. My main intention is to be able to use pruned loss, so i'm adding the **_calc_k2_transducer_pruned_loss**  method inside ESPnetASRModel class. Hope that works!

**Comment by b-flo** [comment]
> Hi @b-flo thanks for your reply. I think the config file you mentioned uses ESPnetASRModel and not ESPnetASRTransducerModel. In fact, most of the recipes are configured to use ESPnetASRModel class. To use ESPnetASRTransducerModel class i followed instructions mentioned here: https://github.com/espnet/espnet/blob/master/doc/espnet2_tutorial.md

Oh, you're right. There was a confusion on my part. AFAIK the only available config for this version in ESPnet is [here](https://github.com/espnet/espnet/pull/4698). Sorry about that!

> I created a config file that ESPnetASRTransducerModel class can accept but didn't get same results as using the ESPnetASRModel class on the same dataset.

I can't comment without additional logs or info but performance should be about the same for both versions (not accounting for k2 pruned loss). Is it not the case with vanilla setup?

> Right now, I'm modifying ESPnetASRModel class to accept k2 pruned loss parameter. My main intention is to be able to use pruned loss, so i'm adding the _calc_k2_transducer_pruned_loss method inside ESPnetASRModel class. Hope that works!

You may need to modify some parts, but yes, it can also works for `ESPnetASRModel`. You can refer to [this PR](https://github.com/espnet/espnet/pull/5268)
